lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/ProcessJourneyEventHandler.java

```diff
@@ -413,4 +417,33 @@ private void sendSubJourneyStartAuditEvent(
                         new AuditExtensionSubjourneyType(journeyType),
                         new AuditRestrictedDeviceInformation(deviceInformation)));
     }
+
+    private boolean isPageState(JourneyState journeyState)
+            throws StateMachineNotFoundException, UnknownStateException {
+        StateMachine stateMachine = stateMachines.get(journeyState.subJourney());
+        if (stateMachine == null) {
+            throw new StateMachineNotFoundException(
+                    String.format(
+                            "State machine not found for journey type: '%s'",
+                            journeyState.subJourney()));
+        }
+        return stateMachine.isPageState(journeyState);
+    }
+
+    private BasicState journeyStateToBasicState(JourneyState journeyState) {
```

Comment:
I think it returns a basic staate because it always will, given the context it's used in - it's only used on a journeyState we've already checked is a BasicState.

Reply:
I think I meant - does the caller actually care that this is a Basic State?

Reply:
I think not - will get it to return the interface